File: JavaScript/Array/L_relativeSortArr.js
var relativeSortArray = function (a, b) {
    let ans = [];
    let f = [];
    for (let i = 0; i < 1001; i++) f.push(0);
    for (let i = 0; i < a.length; i++) f[a[i]]++;
    for (let i = 0; i < b.length; i++) {
        while (f[b[i]]-- > 0) {
            ans.push(b[i]);
        }
    }
    for (let i = 0; i < 1001; i++) {
        while (f[i]-- > 0) {
            ans.push(i);
        }
    }
    return ans;
};

File: cpp/Array/L_relativeSortArr.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> relativeSortArray(vector<int> &a, vector<int> &b)
{
    vector<int> tot(1001);
    for (auto i : a)
        tot[i]++;
    vector<int> ans;
    for (auto i : b)
    {
        if (tot[i])
        {
            while (tot[i]-- > 0)
                ans.emplace_back(i);
        }
    }
    for (int i = 0; i < 1001; i++)
    {
        if (tot[i])
        {
            while (tot[i]-- > 0)
                ans.emplace_back(i);
        }
    }
    return ans;
}